src/MongoDB.Driver.Core/Core/Compression/SnappyCompressor.cs

```diff
@@ -27,28 +27,89 @@ internal class SnappyCompressor : ICompressor
     {
         public CompressorType Type => CompressorType.Snappy;
 
+        /* Declared in header order as defined in the spec:
+         * https://github.com/mongodb/specifications/blob/7461d48df580b2e32b33fdc01188c86fc79773c1/source/compression/OP_COMPRESSED.rst#op_compressed*/
+        private const int MessageLengthSize = sizeof(int);
+        private const int RequestIdSize = sizeof(int);
+        private const int ResponseToSize = sizeof(int);
+        private const int OpCodeSize = sizeof(int);
+        private const int OriginalOpCodeSize = OpCodeSize;
+        private const int UncompressedSizeSize = sizeof(int);
+        private const int CompressorTypeSize = sizeof(byte); // compressorId in spec
+
+        /// <summary>
+        /// Compresses the message  in <paramref name="input"/>, writing the compressed message to
+        /// <paramref name="output"/>.
+        /// </summary>
+        /// <param name="input"> The input stream. Stream MUST HAVE read the exactly up to and including the
+        /// originalOpCode.</param>
+        /// <param name="output">The output stream. Stream MUST HAVE written the header exactly up to and including
+        /// compressorType aka compressorId.</param>
+        /// <remarks>
+        /// The precondition for <paramref name="input"/> mirrors the precondition for <see cref="Decompress"/>'s output
+        /// precondition.
+        /// The precondition for <paramref name="output"/> mirrors the precondition for <see cref="Decompress"/>'s input
+        /// precondition.
+        /// </remarks>
```

Comment:
These comments might no longer be correct depending on what changes you make below.

Reply:
They are no longer preconditions. They only became preconditions when you were seeking backwards into the stream to peek at the data that you assumed had been planted there.

I would like to see these comments removed.

Reply:
Think of it this way: if we had a file called `SnappyCompressorTests` (not sure why we don't) you would NOT need to write any headers to the stream before calling the `Compress` method.

So this is NOT a precondition.

Reply:
Applied changes as discussed offline!